sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -368,6 +373,98 @@ public virtual async Task<Response<UploadBlobResult>> UploadBlobAsync(Stream str
+
+                uploadChunkResult = async ?
+                    await _blobRestClient.UploadChunkAsync(location, stream, cancellationToken: cancellationToken).ConfigureAwait(false) :
+                    _blobRestClient.UploadChunk(location, stream, cancellationToken: cancellationToken);
+
+                return new ChunkedUploadResult(digest,
+                    uploadChunkResult.Headers.Location,
+                    length);
+            }
+
+            byte[] buffer = new byte[maxChunkSize];
+            var ranges = GetRanges(length, maxChunkSize);
+            using SHA256 sha256 = SHA256.Create();
+
+            for (int i = 0; i < ranges.Count; i++)
+            {
+                var range = ranges[i];
+
+                location = uploadChunkResult?.Headers.Location ?? location;
+                var chunkLength = async ?
+                    await stream.ReadAsync(buffer, 0, (int)range.Length, cancellationToken).ConfigureAwait(false) :
+                    stream.Read(buffer, 0, (int)range.Length);
+
+                if (i < ranges.Count - 1)
+                {
+                    sha256.TransformBlock(buffer, 0, chunkLength, buffer, 0);
+                }
+                else
+                {
+                    sha256.TransformFinalBlock(buffer, 0, chunkLength);
+                    digest = "sha256:" + OciBlobDescriptor.BytesToString(sha256.Hash);
+                }
+
+                using (Stream chunk = new MemoryStream(buffer, 0, chunkLength))
+                {
+                    uploadChunkResult = async ?
+                        await _blobRestClient.UploadChunkAsync(location, chunk, GetContentRangeString(range), chunkLength.ToString(CultureInfo.InvariantCulture), cancellationToken).ConfigureAwait(false) :
+                        _blobRestClient.UploadChunk(location, chunk, GetContentRangeString(range), chunkLength.ToString(CultureInfo.InvariantCulture), cancellationToken);
+                }
+            }
+
+            return new ChunkedUploadResult(digest, uploadChunkResult.Headers.Location, length);
+        }
+
+        /// <summary>
+        /// ACR has a non-standard use of the Content-Range header in the PATCH chunked
+        /// upload request. This converts range to the format used by this API,
+        /// <see href="https://docs.docker.com/registry/spec/api/#patch-blob-upload"/> for details.
+        /// </summary>
+        /// <param name="range">The range indicating the size of a chunk.</param>
+        /// <returns>A string formatted to suit the Content-Range header.</returns>
+        private static string GetContentRangeString(HttpRange range)
+        {
+            var endRange = "";
+            if (range.Length.HasValue && range.Length != 0)
+            {
+                endRange = (range.Offset + range.Length.Value - 1).ToString(CultureInfo.InvariantCulture);
+            }
+
+            return FormattableString.Invariant($"{range.Offset}-{endRange}");
```

Comment:
Should we use the Core [HttpRange](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/HttpRange.cs#L15) type?

Reply:
Nevermind, I see the ref docs that explain this doesn't use the standard format for content-range.